fastlane_bot/helpers/poolandtokens.py

```diff
@@ -476,6 +499,66 @@ def decimal_converter(idx):
                 "descr": self.descr,
                 "params": self._params,
             }
+
+            strategy_typed_args += [typed_args]
+
+        #### MODIFICATION LOGIC HERE >>>>>
+        # Only overlapping strategies are selected for modification
+        if len(strategy_typed_args) == 2:
+            
+            is_overlapping = False
+            pmarg_threshold = Decimal("0.01") # 1%  # WARNING using this condition alone can included stable/stable pairs incidently
+
+            # evaluate that the marginal prices are within the pmarg_threshold
+            pmarg0, pmarg1 = [x['p_marg'] for x in strategy_typed_args]
+            pmarg0_inv = 1/pmarg0  # one of the orders must be flipped since prices are always dy/dx - but must flip same geomean_pmarg later
+            percent_component = pmarg_threshold * max(pmarg0_inv, pmarg1)
+            percent_component_met = abs(pmarg0_inv - pmarg1) <= percent_component
+
+            # overlapping strategies by defintion cannot have A=0 i.e. there must be no limit orders
+            no_limit_orders = (strategy_typed_args[0]['is_limit_order'] == False) and (strategy_typed_args[1]['is_limit_order'] == False)
+
+            # evaluate if the price boundaries pa/pb overlap at one end # TODO check logic and remove duplicate logic if necessary
+            prices_overlap = check_overlap(strategy_typed_args[0]['pa'], strategy_typed_args[0]['pb'], strategy_typed_args[1]['pa'], strategy_typed_args[1]['pb'])
+            # if (percent_component_met and no_limit_orders) and not prices_overlap:
+            #     print(percent_component_met, no_limit_orders, prices_overlap)
+            #     print(strategy_typed_args)
+
+            # if the threshold is met and neither is a limit order and prices overlap then likely to be overlapping
+            is_overlapping = percent_component_met and no_limit_orders and prices_overlap
```

Comment:
A strategy with one limit order and one range order can still constitute an overlapping strategy.
In fact, even a strategy with two limit orders can still constitute an overlapping strategy (despite of it being senseless on behalf of its owner to create such strategy).

Reply:
Indeed but outside of the scope of this work so not required here